packages/blade/src/components/FileUpload/FileUpload.web.tsx

```diff
@@ -0,0 +1,367 @@
+                  </Text>
+                  <SelectorInput
+                    id={inputId}
+                    hoverTokens={getFileUploadInputHoverTokens()}
+                    isChecked={false}
+                    isDisabled={isDisabled}
+                    inputProps={{
+                      name,
+                      type: 'file',
+                      onChange: handleInputChange,
+                      multiple: isMultiple,
+                      required: isRequired,
+                      disabled: isDisabled,
+                      accept,
+                      onBlur: () => setIsActive(false),
+                      ...accessibilityProps,
+                    }}
+                    ref={ref}
+                  />
+
+                  <Box
+                    display="flex"
+                    justifyContent="center"
+                    alignItems="center"
+                    flexDirection={{ xs: 'column', s: 'row' }}
+                    borderRadius="small"
+                  >
+                    <Box
+                      display="flex"
+                      flexDirection="row"
+                      alignItems="center"
+                      borderBottomColor={
+                        isDisabled
+                          ? 'surface.border.primary.muted'
+                          : 'surface.border.primary.normal'
+                      }
+                      borderBottomWidth="thin"
+                    >
+                      <Text
+                        color={
+                          isDisabled
+                            ? fileUploadColorTokens.link.disabled
+                            : fileUploadColorTokens.link.default
+                        }
+                      >
+                        Upload
+                      </Text>
+                    </Box>
+                  </Box>
+                </Box>
+              </StyledFileUploadWrapper>
+            </BaseBox>
+          </SelectorLabel>
+          {isOneFileSelectedWithSingleUpload && (
+            <BaseBox key={selectedFiles[0].id} marginBottom="spacing.5">
+              <FileUploadItem
+                file={selectedFiles[0]}
+                onRemove={() => {
+                  const newFiles = selectedFiles.filter(({ id }) => id !== selectedFiles[0].id);
+                  setSelectedFiles(newFiles);
+                  onRemove?.({ file: selectedFiles[0] });
+                }}
+                onCancel={() => {
+                  const newFiles = selectedFiles.filter(({ id }) => id !== selectedFiles[0].id);
+                  setSelectedFiles(newFiles);
+                  onCancel?.({ file: selectedFiles[0] });
+                }}
+                onPreview={onPreview}
+              />
+            </BaseBox>
+          )}
+        </BaseBox>
+      </BaseBox>
+      {/* the magic number 136 is basically max-width of label i.e 120 and then right margin i.e 16 which is the spacing between label and input field */}
```

Comment:
Move it constant maybe?

Reply:
It's used in Baseinpu & OTPInput as well, let's refactor this separately.